packages/tools/src/utilities/cine/playClip.ts

```diff
@@ -262,4 +296,143 @@ function _stopClipWithData(playClipData) {
+}
+
+function _createStackViewportCinePlayContext(
+  viewport: StackViewport
+): CINETypes.CinePlayContext {
+  const imageIds = viewport.getImageIds();
+
+  return {
+    get numScrollSteps(): number {
+      return imageIds.length;
+    },
+    get currentStepIndex(): number {
+      return viewport.getTargetImageIdIndex();
+    },
+    get frameTimeVectorEnabled(): boolean {
+      // It is always in acquired orientation
+      return true;
+    },
+    scroll(delta: number): void {
+      scroll(viewport, { delta, debounceLoading: debounced });
+    },
+  };
+}
+
+function _createVolumeViewportCinePlayContext(
+  viewport: VolumeViewport,
+  volume: Types.IImageVolume
+): CINETypes.CinePlayContext {
+  const actorEntry = viewport.getDefaultActor();
+
+  if (!actorEntry) {
+    console.warn('No actor found');
+  }
+
+  const volumeId = actorEntry.uid;
+  const cachedScrollInfo = {
+    viewPlaneNormal: vec3.create(),
+    scrollInfo: null,
+  };
+
+  const getScrollInfo = () => {
+    const camera = viewport.getCamera();
+    const updateCache =
+      !cachedScrollInfo.scrollInfo ||
+      !vec3.equals(camera.viewPlaneNormal, cachedScrollInfo.viewPlaneNormal);
+
+    // Number of steps would change only after rotating the volume so it
+    // caches the result and recomputes only when necessary. Until it is
+    // rotated the current frame is updated locally
+    if (updateCache) {
+      const scrollInfo = csUtils.getVolumeViewportScrollInfo(
+        viewport,
+        volumeId
+      );
+
+      cachedScrollInfo.viewPlaneNormal = camera.viewPlaneNormal;
+      cachedScrollInfo.scrollInfo = scrollInfo;
+    }
+
+    return cachedScrollInfo.scrollInfo;
+  };
+
+  return {
+    get numScrollSteps(): number {
+      return getScrollInfo().numScrollSteps;
+    },
+    get currentStepIndex(): number {
+      return getScrollInfo().currentStepIndex;
+    },
+    get frameTimeVectorEnabled(): boolean {
+      const camera = viewport.getCamera();
+      const volumeViewPlaneNormal = volume.direction
+        .slice(6, 9)
+        .map((x) => -x) as Types.Point3;
+      const dot = vec3.dot(volumeViewPlaneNormal, camera.viewPlaneNormal);
+
+      // Check if the volume is in acquired orientation
+      // it may be flipped or rotated in plane
+      return glMatrix.equals(dot, 1);
+    },
+    scroll(delta: number): void {
+      getScrollInfo().currentStepIndex += delta;
+      scroll(viewport, { delta });
+    },
+  };
+}
+
+function _createDynamicVolumeViewportCinePlayContext(
+  volume: Types.IDynamicImageVolume
+): CINETypes.CinePlayContext {
+  return {
+    get numScrollSteps(): number {
+      return volume.numTimePoints;
+    },
+    get currentStepIndex(): number {
+      return volume.timePointIndex;
+    },
+    get frameTimeVectorEnabled(): boolean {
+      // Looping throught time does not uses frameTimeVector
+      return false;
+    },
+    scroll(delta: number): void {
+      volume.timePointIndex += delta;
```

Comment:
maybe put a comment that this is a setter.

Reply:
Done